pkg/kube/cluster-init.sh

```diff
@@ -70,17 +103,85 @@ get_default_intf_IP_prefix() {
         awk -v new_ip="$ip_prefix" '{gsub("IPAddressReplaceMe", new_ip)}1' /etc/multus-daemonset.yaml > /tmp/multus-daemonset.yaml
 }
 
+# kubernetes's name must be lower case and '-' instead of '_'
+convert_to_k8s_compatible() {
+        echo "$1" | tr '[:upper:]_' '[:lower:]-'
+}
+
+wait_for_device_name() {
+        logmsg "Waiting for DeviceName from controller..."
+        EdgeNodeInfoPath="/persist/status/zedagent/EdgeNodeInfo/global.json"
+        while true; do
```

Comment:
the `if` condition can just be put here, can't it?

Does it perhaps make sense to use inotify?

Reply:
this was addressed in the broken up pr as well.